godep/device.go

```diff
@@ -124,11 +130,24 @@ type DeviceListResponse struct {
 // details on a set of devices.
 // See https://developer.apple.com/documentation/devicemanagement/get_device_details
 func (c *Client) DeviceDetails(ctx context.Context, name string, serials ...string) (*DeviceListResponse, error) {
-	req := struct {
-		Devices []string `json:"devices"`
-	}{
-		Devices: serials,
-	}
+	req := DeviceListRequest{Devices: serials}
 	resp := new(DeviceListResponse)
 	return resp, c.do(ctx, name, http.MethodPost, "/devices", req, resp)
 }
+
+// DeviceStatusResponse corresponds to the Apple API "DeviceStatusRepoonse" structure.
+// See https://developer.apple.com/documentation/devicemanagement/devicestatusresponse
+type DeviceStatusResponse struct {
+	Devices map[string]string `json:"devices"`
+}
+
+// DisownDevices uses the Apple "Disown Devices" API endpoint to disclaim
+// ownership of device serial numbers.
+// WARNING: This will permanantly remove devices from the ABM/ASM/ABE instance.
```

Comment:
Can disowned devices be brought back with the new provisional DEP for macOS stuff? ala https://support.apple.com/guide/apple-configurator/intro-apd4015ec300/1.1/ios/17.0

Reply:
I'm told they can, yeah. I guess "permanent" here is more of a warning than a fact here, heh. :)